cpp_guidelines/.clang-format

```diff
@@ -0,0 +1,107 @@
+---
+Language: Cpp
+Standard: c++17
+DisableFormat:   false
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: true
+AlignConsecutiveBitFields: true
+AlignConsecutiveDeclarations: false
+AlignConsecutiveMacros: true
+AlignEscapedNewlines: Right
+AlignOperands: true
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: false
+AllowAllConstructorInitializersOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: Never
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortEnumsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: All
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: Yes
+BinPackArguments: false
+BinPackParameters: false
+BitFieldColonSpacing: Both
+BreakBeforeBinaryOperators: All
+BreakBeforeBraces: Allman
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: BeforeComma
+BreakInheritanceList: BeforeComma
+BreakStringLiterals: true
+ColumnLimit:     79
```

Comment:
I know 80 chars is super popular. I just want to note that I really prefer longer widths, because this fits better how monitors look like nowadays ;) I have seen widths of 120 used every now and then. I privately use 160. That is still narrow enough to get two files next to each other on a normal 16:9 monitor and have you file explorer from your IDE on the side.

However, if everybody is super keen on having their 80 char vim in a terminal than go ahead.

Reply:
This limitation is mostly motivated by three way diffs. We had this discussion in the past and collected arguments for 80 chars https://github.com/ComputationalRadiationPhysics/contributing/issues/2#issuecomment-159303617

Reply:
I think 80 limit brings much more harm than use.

Reply:
> I think 80 limit brings much more harm than use.

👍 

automatic line wrapping should make 3 way-diffs as easy with higher line-lengths as with 80. I personally prefer `0`.

Reply:
I think I never did a 3 way diff or merge. Someone needs to show me at some point!

Maximized window on a 16:9 1440p monitor:
In VS and I can get around 296 in one line when I still have the file system explorer open on the side. Otherwise I get 345 chars. Notepad++ gives me around 300 chars. Powershell and bash get to 318 chars.

I conclude that at least 100 chars should be possible and still be able to have 3 windows side by side. With 2 windows side by side we can get up to 150 chars.

Reply:
**VOTE**
current (80) | 120 |
--------|-----------|
 0      |   1

Monitors are bigger nowadays and a short column limit expands the code vertically a lot, increasing the need to scroll around.
120 is large enough to fit 2 files next to each other and still have some space for e.g. a file explorer window in your IDE. 140 would also work, 160 seems too excessive to me, since then 2 files next too each other starts to no longer display nicely. On the contrary, 100 seems too little for me, but would probably work best for 3 files per monitor. But 3 files does not work nicely with snapping windows to the left and right of your monitor (cf. Windows Aero Snap) ;)

Reply:
current (80) | 120
-- | --
0 | 2

I agree with 120. We also need to think of 1080p resolution. There a vim `vsplit` gives me 130 columns, 120 leaves 20 columns for an eventual file browser column.

Reply:
I agree with @jkelling, 1080p should be the baseline. My emacs breaks at 118 columns at two horizontal windows. So 120 is fine for me.

```
current (80) | 120 |
-------------|-----|
 0           |   3   
```

Reply:
current (80) | 120 |
----------------|-------|
 0 | 4 |

Reply:
current (80) | 120
-- | --
0 | 4

Reply:
current (80) | 120
-- | --
0 | 5

Reply:
current (80) | 120 |
--------|-----------|
 1      |   5

Reply:
current (80) | 120
-- | --
1 | 6

Reply:
current (80) | 120
-- | --
1 | 7

Reply:
Reminder: Voting will be closed on 2020-09-26 at 12:22 HZDR time.

Reply:
current (80) | 120 |
--------|-----------|
 1      |   8

Reply:
current (80) | 120
-- | --
2 | 8

Reply:
Voting closed, the winner is `120` (or rather `119` to account for the extra symbol added by `diff`).